### ISSUE TITLE
[Custom Handler] Live metrics memory is wrong

### ISSUE THREAD

**Issue by jdubois** [Azure/azure-functions-host#6184]
This is linked to #6172

Using the same https://github.com/jdubois/azure-native-spring-function example, the memory usage on my machine is 32 Mb (which is expected).

- When using the "~2" version of the Functions runtime, memory usage is shown as "0 mb", which is obviously wrong.
- When using the "~3" version of the Functions runtime, memory usage is shown as "700 mb". This seems way too much, so I would like to know what exactly causes that. Are you monitoring more than just the process that was created?

This is running on Linux with the consumption plan, I'm going to do more tests, using Windows, to see if there's a difference.

**Comment by yojagad** [comment]
Thanks for opening an issue on this, we'll take a look.

**Comment by v-anvari** [comment]
Hi @jdubois , Can we know if you are seeing this issue still and requires further follow up

**Comment by jdubois** [comment]
![Screen Shot 2021-10-19 at 14 46 42](https://user-images.githubusercontent.com/316835/137912320-dece7ac5-5e76-4adf-a391-679ba597eecb.png)
Yes here is a screenshot I just did: as you can see, I had 7 executions (which is correct), but this used 0 bytes of memory, which is obviously wrong. I also wonder how the customers are billed in this case, as this depends on the memory used.

**Comment by v-anvari** [comment]
Thank you for the confirmation. I shall further follow up with the team on this issue. Can you confirm if this is specific to any OS?

**Comment by jdubois** [comment]
This is a custom handler, so yes it's specific to Linux. It's running inside a container.

**Comment by fabiocav** [comment]
@ehamai where does the portal gets these metrics from? 

@balag0 looks like this is setup with a custom container, would you expect those metrics to be available with that scenario? Are you aware of any issues with that?

@madelinegordon would you please be able to validate whether other SKUs work as expected with custom handlers?

**Comment by balag0** [comment]
Custom handlers are not considered custom containers as the image is still not configurable. so there should be no difference from other apps.
 +@CooperLink has been investigating some of these issues recently.

**Comment by CooperLink** [comment]
I have been investigating some similar issues in Linux Consumption.  The issue that I am working on doesn't seem to apply here though.  The behavior that I am seeing in the known issue is that no metrics at all make it to portal.  In this case, Julien is seeing executions but not any memory events.  
In my opinion, the custom container trigger should not introduce any problems related to billing like Bala said.  A few characteristics that I am interested in for the function site.  We know that the known issue arises when a function app sticks to a single worker for a long duration of time in Linux Consumption. Could you share the function app name @jdubois ?

@fabiocav I am fairly certain that Portal populates those metrics from MDM.  Metrics like these flow from the Metering Agent running in Atlas Nodes to the MDM service.

**Comment by jdubois** [comment]
@CooperLink the function name is called "judubois-native-functions", and you can test it using:
```bash
curl https://judubois-native-functions.azurewebsites.net/api/hello -d '{"name": "Julien"}' -H "Content-Type: application/json"
```

If you want more information (like my subscription ID, which shouldn't be share publicly), please ping me internally on Teams (my Microsoft ID is "judubois").

**Comment by fabiocav** [comment]
@CooperLink @yojagad (in case you spent time investigating) do you have any updates on this?

I believe @mathewc looked at a similar issue in the past, so pinging him here in case he has additional information to share.

**Comment by soninaren** [comment]
@CooperLink , @yojagad Can you share an update if there was investigation performed?

**Comment by liliankasem** [comment]
@CooperLink , @yojagad Can you share an update if there was investigation performed? 😄

**Comment by fabiocav** [comment]
@jdubois @CooperLink @yojagad can you confirm whether this is still an issue and requires additional follow up?

**Comment by jdubois** [comment]
@fabiocav yes I confirm nothing has changed. Here's a new screenshot I did right now, it's the same issue as 2 years ago.

![Screen Shot 2022-03-23 at 21 50 19](https://user-images.githubusercontent.com/316835/159793442-f194e606-67a2-4029-9b32-9b43c3d53e43.png)

**Comment by jdubois** [comment]
Hey, I’m the author and I gave feedback each time it was asked.
You cannot close the issue like that, please keep it opened.
